module4/projects/solo_stretch_tech/index.md

```diff
@@ -3,14 +3,137 @@ title: Solo Stretch Tech
+- **Rate Limiting:** Consider using the gem called “rack-attack” and test to demonstrate it’s working! Students should be able to discuss why rate limiting is an effective feature. Ideally, be able to demonstrate it live using a small script (either live or in production).
+- **Microservice:** Students should create a separate microservice (separate from their BE API app) using Sinatra or a different framework, and be able to speak to the pros/cons of that framework when compared to Rails.
+- **JWT (JSON Web Tokens):** Use the jwt gem for stateless authentication alongside username/password login. Students should explain its benefits and show token expiration and verification
+<!-- **I used chatGPT for the following list items, please change as needed** -->
+- **GraphQL:** Use GraphQL for querying and mutating data. Students should be able to explain the differences between GraphQL and REST. (This would lead to some changes to how you make requests from the frontend, too!)
+- **Pagination:** Implement pagination to limit the amount of data returned in API responses. Students can use strategies like cursor-based pagination or offset-based pagination. They should discuss why pagination is important for performance and scalability.
+- **[Security Vulnerability Scanner](https://github.com/presidentbeef/brakeman):** Use Brakeman, a static analysis tool, to scan the Rails application for potential security vulnerabilities. Students should be able to explain what Brakeman does, why it’s useful, and how it integrates into the development workflow. 
+- **OpenAPI Documentation:** Use OpenAPI Specification to create comprehensive, interactive API documentation. Students should document all major API endpoints (requests, responses, and parameters) and integrate the documentation using tools like Swagger UI or Redoc.
+- **[Circuit Breaker Design Pattern](https://makisushi.io/posts/asd):** Implement the Circuit Breaker design pattern to handle potential failures when calling external APIs or services. Students should use a gem like stoplight or breaker to demonstrate how the circuit breaker prevents cascading failures and improves app resilience. 
+- **Observability:** Integrate an observability tool (e.g., New Relic, Skylight, Honeybadger) to monitor and analyze the application’s performance, errors, and resource usage. Students should configure the tool to track response times, database queries, and bottlenecks.
+</section>
+
+## Timeline + Deliverables
+
+At the end of the days listed below, you will need to turn in certain deliverables. Your cohort calendar specifies the day of the stretch project so you can easily align when each set of deliverables are due. If you fall behind, talk to your instructor as soon as possible. We want to mimic what communication would look like on the job if you needed more time to complete your assigned tasks.
+
```

Comment:
Do the deliverables seem like over-management or supportive to get them to focus on the right things? I want them to do minimal coding and focus on planning. I figured having super outlined deliverables could help with that?

Reply:
I think this sounds like a good place to start. Yes they did a lot of this in consultancy, but it was with a group. I think the handholding will be helpful in addition to keeping them accountable.